src/d_main.c

```diff
@@ -303,20 +313,17 @@ void D_Display (void)
   inhelpscreensstate = inhelpscreens;
   oldgamestate = wipegamestate = gamestate;
 
-  if (gamestate == GS_LEVEL && automapactive)
+  if (gamestate == GS_LEVEL && automapactive && automapoverlay)
     {
       AM_Drawer();
+      ST_Drawer(scaledviewheight == 200, redrawsbar);
```

Comment:
To achieve the necessary drawing order, I added this `ST_Drawer()` call right here.

Reply:
Alright.